test/integration/replicaset/replicaset_test.go

```diff
@@ -930,3 +933,59 @@ func TestFullyLabeledReplicas(t *testing.T) {
 		t.Fatalf("Failed to verify only one pod is fully labeled: %v", err)
 	}
 }
+
+func getOrphanOptions() *metav1.DeleteOptions {
+	var trueVar = true
+	return &metav1.DeleteOptions{OrphanDependents: &trueVar}
+}
+
+func TestReplicaSetsExtensionsV1beta1DefaultGCPolicy(t *testing.T) {
+	s, closeFn, rm, informers, c := rmSetup(t)
+	defer closeFn()
+	ns := framework.CreateTestingNamespace("test-default-gc-extensions", s, t)
+	defer framework.DeleteTestingNamespace(ns, s, t)
+	stopCh := runControllerAndInformers(t, rm, informers, 0)
+	defer close(stopCh)
+
+	rs := newRS("rs", ns.Name, 2)
+	//rs.Finalizers = []string{"kube.io/dummy-finalizer"}
+	rss, _ := createRSsPods(t, c, []*v1beta1.ReplicaSet{rs}, []*v1.Pod{})
+	rs = rss[0]
+	waitRSStable(t, c, rs)
+
+	// Verify RS creates 2 pods
+	podClient := c.CoreV1().Pods(ns.Name)
+	pods := getPods(t, podClient, labelMap())
+	if len(pods.Items) != 2 {
+		t.Fatalf("len(pods) = %d, want 2", len(pods.Items))
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		w, err := c.ExtensionsV1beta1().ReplicaSets(ns.Name).Watch(metav1.SingleObject(metav1.ObjectMeta{Name: rs.Name}))
+		if err != nil {
+			t.Fatalf("Failed watch: %v", err)
+		}
+		_, err = watch.Until(20*time.Second, w, func(e watch.Event) (bool, error) {
+			switch o := e.Object.(type) {
+			case *v1beta1.ReplicaSet:
+				if slice.ContainsString(o.Finalizers, metav1.FinalizerOrphanDependents, nil) {
+					return true, nil
+				}
+			}
+			return false, fmt.Errorf("unexpected watch event containing object %#q", e)
+		})
+		if err != nil {
+			t.Fatalf("Failed to find finalizer orphan: %v", err)
```

Comment:
This is the case that the test is hitting:

```
--- FAIL: TestReplicaSetsExtensionsV1beta1DefaultGCPolicy (8.48s)
	replicaset_test.go:981: Failed to find finalizer orphan: unexpected watch event containing object {`ADDED` `&ReplicaSet{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:rs,GenerateName:,Namespace:test-default-gc-extensions,SelfLink:/apis/extensions/v1beta1/namespaces/test-default-gc-extensions/replicasets/rs,UID:362c1d7b-cbae-11e7-9852-0242ac110002,ResourceVersion:12652,Generation:1,CreationTimestamp:2017-11-17 15:44:35 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{foo: bar,},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,},Spec:ReplicaSetSpec{Replicas:*2,Selector:&k8s_io_apimachinery_pkg_apis_meta_v1.LabelSelector{MatchLabels:map[string]string{foo: bar,},MatchExpressions:[],},Template:k8s_io_api_core_v1.PodTemplateSpec{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{foo: bar,},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,},Spec:PodSpec{Volumes:[],Containers:[{fake-name fakeimage [] []  [] [] [] {map[] map[]} [] nil nil nil /dev/termination-log File Always nil false false false}],RestartPolicy:Always,TerminationGracePeriodSeconds:*30,ActiveDeadlineSeconds:nil,DNSPolicy:ClusterFirst,NodeSelector:map[string]string{},ServiceAccountName:,DeprecatedServiceAccount:,NodeName:,HostNetwork:false,HostPID:false,HostIPC:false,SecurityContext:&PodSecurityContext{SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:nil,},ImagePullSecrets:[],Hostname:,Subdomain:,Affinity:nil,SchedulerName:default-scheduler,InitContainers:[],AutomountServiceAccountToken:nil,Tolerations:[],HostAliases:[],PriorityClassName:,Priority:nil,},},MinReadySeconds:0,},Status:ReplicaSetStatus{Replicas:2,FullyLabeledReplicas:2,ObservedGeneration:1,ReadyReplicas:0,AvailableReplicas:0,Conditions:[],},}`}
FAIL
FAIL	k8s.io/kubernetes/test/integration/replicaset	56.504s
```

The log extraction in gubernator seems to get confused, so you need to go to the raw `.stdout` files by clicking the [artifacts](http://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/55148/pull-kubernetes-unit/66293/artifacts/) link to find the real test output.